qualcommax: ipq807x: add support for Spectrum SAX1V1K

gotofbi opened the pull request:
Spectrum SAX1V1K is a AX WIFI router with 3 1G and 1 2.5G ports. 
The router is provided to Spectrum customers.
It is OEM of Askey RT5010W
https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923

It continues the original work by @MeisterLone to get this device supported.

### Specifications:
```
  •  CPU: Qualcomm IPQ8072A Quad core Cortex-A53 2.2GHz
  •  RAM: 2048MB of DDR3
  •  Storage: 1024MB eMMC
  •  Ethernet: 3x 1G RJ45 ports (QCA8075) + 1 2.5G Port (QCA8081)
  •  WLAN:
     • 2.4GHz: Qualcomm QCN5024 4x4 802.11b/g/n/ax 1174 Mbps PHY rate
     • 5GHz: Qualcomm QCN5054 4x4 802.11a/b/g/n/ac/ax 2402 PHY rate

  •  LED: 1 gpio-controlled dual color led (blue/red)
  •  Buttons: 1x reset
  •  Power: 12V DC jack
``` 


### Notes:
```
  •  This commit adds only single partition support, that means
     sysupgrade is upgrading the current rootfs partition.
  •  Installation can be done by serial connection only.
  •  A poulated serial header is onboard
     https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923/6
  •  RX/TX is working, u-boot bootwait is active, secure boot is enabled.
```

### Installation Instructions:

	**Most part of the installation is performed from an initramfs image.**

	Boot initramfs : Using serial connection
	1. Boot up the device and wait till it displays "VERIFY_IB: Success. verify IB ok"
	2. Once that message appears, 
	    login with username 'root'
	    password serial number of your router in uppercase.
	3. Use vi to paste the 'open.sh' script from @MeisterLone github on your device
	    https://github.com/gotofbi/Qualcommax_NSS_sax1v1k/blob/main/open.sh
	4. chmod 755 open.sh
	5. ./open.sh
	6. Set your ip to 192.168.0.1
	7. Run a TFTP server and host the initramfs image on the TFTP server and name it "recovery.img"
	8. Reboot device. On boot it will try TFTP. 

	Install OpenWrt from initramfs image:
	1. Use SCP (or other way) to transfer OpenWrt factory image
	2. Connect to device using SSH (on a LAN port)
	3. Flash firmware: sysupgrade
	   # sysupgrade -n -v /tmp/openwrt_sysupgrade.bin
	4. Set U-boot env variable: bootcmd
	   # fw_setenv bootcmd "run fix_uboot; run setup_and_boot"
	5. Reboot the device
	   # reboot
	6. Once device is booted, residue of previous firmware will prevent openwrt to work properly.
	    Factory Reset is MUST required
	    # Once serial console is displaying to login, hold reset button for 10 sec
	7. Now everything should be operational.

        Note: this PR adds only single partition support, that means sysupgrade is
              upgrading the current rootfs partition

### QCA Firmware PR
https://github.com/openwrt/firmware_qca-wireless/pull/36

motolav commented:
The Dynalink router is literally a Askey RT5010W(4 1g + 1 2.5g) with different branding, it'd be better to use the model name that's on the case

gotofbi commented:
> The Dynalink router is literally a Askey RT5010W(4 1g + 1 2.5g) with different branding, it'd be better to use the model name that's on the case if this firmware is for the device 'Spectrum SAX1V1K'

You are absolutely correct and I actually considered about how to name this.

Dynalink DL-WRX36 is actually Askey RT5010W-D350
https://fcc.report/FCC-ID/H8NRT5010W-D350
4 1g + 1 2.5g, USB, 256gb nand

This PR is for Spectrum SAX1V1K which is actually Askey RT5010W-D187
https://fcc.report/FCC-ID/H8NSAX1V1K
3 1g + 1 2.5g, no USB, 1tb emmc

This is ISP provided router.
So if I follow that, the branding will become "Spectrum SAX1V1K" which I thought its kinda odd.
"Spectrum" is not device manufacture but its ISP.

If you still think "Spectrum SAX1V1K" is more appropriate naming, I will make an update.

motolav commented:
If "Askey RT5010W-D187" is only sold as "Spectrum SAX1V1K" it makes more sense to just use the name people would see on the case. 'Askey RT5010W-D350' is the standard variant for the RT5010W on Askey's site too.

gotofbi commented:
> If "Askey RT5010W-D187" is only sold as "Spectrum SAX1V1K" it makes more sense to just use the name people would see on the case. 'Askey RT5010W-D350' is the standard variant for the RT5010W on Askey's site too.

You are right!
I will update the PR

rmilecki commented on package/firmware/ipq-wifi/Makefile:
```diff
@@ -31,6 +31,7 @@ ALLWIFIBOARDS:= \
 	8devices_mango \
 	arcadyan_aw1000 \
 	buffalo_wxr-5950ax12 \
+	askey_rt5010w-d187 \
```
Please keep alphabetical order (`a` goes before ` b`).

gotofbi replied:
fixed

rmilecki commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-rt5010w-d187.dts:
```diff
@@ -0,0 +1,250 @@
+		compatible = "gpio-leds";
+
+		led_system_blue: system-blue {
+			label = "blue:system";
```
You already have `color`. Please add ` function` and drop `label`.

gotofbi replied:
fixed

rmilecki commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-rt5010w-d187.dts:
```diff
@@ -0,0 +1,250 @@
+};
+
+&sdhc_1 {
+	status = "okay";
```
`status` should be the last property in node. Applies to all `status` properties in this .dts.

gotofbi replied:
Can you explain me further more regarding this?
I referenced dl-wrx36 dts since it is almost similar hardware.
https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-dl-wrx36.dts

SpectreDev1 replied:
Nah you're ok, it's not necessary, this is fine

rmilecki replied:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/dts-coding-style.rst?h=v6.8#n112

"Status is the last information to annotate (...)"

SpectreDev1 commented on target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata:
```diff
@@ -10,6 +10,9 @@ case "$FIRMWARE" in
 "ath11k/IPQ8074/hw2.0/cal-ahb-c000000.wifi.bin")
 	case "$board" in
 	arcadyan,aw1000|\
+		askey,rt5010w-d187)
```
Fix the whitespace here, and this shouldnt be here since you are changing the caldata extraction of another device

SpectreDev1 replied:
move this to the the prpl,haze section, no need to create duplicate of the same command

gotofbi replied:
fixed

SpectreDev1 commented on target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh:
```diff
@@ -44,6 +44,11 @@ platform_pre_upgrade() {
 platform_do_upgrade() {
 	case "$(board_name)" in
 	arcadyan,aw1000|\
+	askey,rt5010w-d187)
```
this shouldnt be here since you are changing the platform upgrade of another device

SpectreDev1 replied:
move this to the the prpl,haze section, no need to create duplicate of the same command

gotofbi replied:
fixed

SpectreDev1 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-rt5010w-d187.dts:
```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/* Copyright (c) 2022, Robert Marko <robimarko@gmail.com> */
```
Are you sure this DTS belongs to robimarko

gotofbi replied:
fixed

SpectreDev1 commented:
Fix your commit it shouldnt have merge branch into main

SpectreDev1 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-rt5010w-d187.dts:
```diff
@@ -0,0 +1,250 @@
+	status = "okay";
+};
+
+&qpic_nand {
```
if you have an EMMC and No nand there isnt a need for this

gotofbi replied:
fixed

SpectreDev1 commented:
> > Fix your commit it shouldnt have merge branch into main
> 
> My bad, looks like I cannot edit branch. Should I create new PR?

Should be able to rebase

gotofbi committed:
ipq807x: add Spectrum SAX1V1K

Spectrum SAX1V1K is a AX WIFI router with 3 1G and 1 2.5G ports.
The router is provided to Spectrum customers.
It is OEM of Askey RT5010W
https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923

It continues the original work by @MeisterLone to get this device supported.

Specifications:
```
  •  CPU: Qualcomm IPQ8072A Quad core Cortex-A53 2.2GHz
  •  RAM: 2048MB of DDR3
  •  Storage: 1024MB eMMC
  •  Ethernet: 3x 1G RJ45 ports (QCA8075) + 1 2.5G Port (QCA8081)
  •  WLAN:
     • 2.4GHz: Qualcomm QCN5024 4x4 802.11b/g/n/ax 1174 Mbps PHY rate
     • 5GHz: Qualcomm QCN5054 4x4 802.11a/b/g/n/ac/ax 2402 PHY rate

  •  LED: 1 gpio-controlled dual color led (blue/red)
  •  Buttons: 1x reset
  •  Power: 12V DC jack
```

Notes:
```
  •  This commit adds only single partition support, that means
     sysupgrade is upgrading the current rootfs partition.
  •  Installation can be done by serial connection only.
  •  A poulated serial header is onboard
     https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923/6
  •  RX/TX is working, u-boot bootwait is active, secure boot is enabled.
```

Installation Instructions:

	**Most part of the installation is performed from an initramfs image.**

	Boot initramfs : Using serial connection
	1. Boot up the device and wait till it displays "VERIFY_IB: Success. verify IB ok"
	2. Once that message appears,
	    login with username 'root'
	    password serial number of your router in uppercase.
	3. Use vi to paste the 'open.sh' script from @MeisterLone github on your device
	    https://github.com/MeisterLone/Askey-RT5010W-D187-REV6/blob/master/Patch/open.sh
	4. chmod 755 open.sh
	5. ./open.sh
	6. Set your ip to 192.168.0.1
	7. Run a TFTP server and host the initramfs image on the TFTP server and name it "recovery.img"
	8. Reboot device. On boot it will try TFTP.

	Install OpenWrt from initramfs image:
	1. Use SCP (or other way) to transfer OpenWrt factory image
	2. Connect to device using SSH (on a LAN port)
	3. Flash firmware: sysupgrade
	   # sysupgrade -n -v /tmp/openwrt_sysupgrade.bin
	4. Set U-boot env variable: bootcmd
	   # fw_setenv bootcmd "run fix_uboot; run setup_and_boot"
	5. Reboot the device
	   # reboot
	6. Once device is booted, residue of previous firmware will prevent openwrt to work properly.
	    Factory Reset is MUST required
	    # Once serial console is displaying to login, hold reset button for 10 sec
	7. Now everything should be operational.

        Note: this PR adds only single partition support, that means sysupgrade is
              upgrading the current rootfs partition

Signed-off-by: Connor Yoon <j_connor@taliaent.com>

gotofbi commented:
I screwed QCA firmware PR and had to create new one.
https://github.com/openwrt/firmware_qca-wireless/pull/36

SpectreDev1 commented on target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh:
```diff
@@ -90,6 +90,7 @@ platform_do_upgrade() {
 		fw_setenv auto_recovery yes
 		nand_do_upgrade "$1"
 		;;
+	spectrum,sax1v1k|\
```
P Q R S, it's in the wrong place needs to be below qnap

gotofbi replied:
Sorry, case sensitive from here as well.

gotofbi replied:
fixed

SpectreDev1 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,241 @@
+/dts-v1/;
+
```
Add your license

gotofbi replied:
added gpl

SpectreDev1 commented on package/boot/uboot-envtools/files/qualcommax_ipq807x:
```diff
@@ -8,6 +8,11 @@ touch /etc/config/ubootenv
 case "$board" in
+spectrum,sax1v1k)
+	part="$(find_mmc_part 0:APPSBLENV)"
+	[ -n "$part" ] && \
```
I'm not 100% sure but ain't this supposed to mmcpart

gotofbi replied:
$part is temp variable.
Like "prpl,haze", i will change it to $mmcpart

gotofbi replied:
fixed and also placed entire case at the end of case blob for better visibility.

SpectreDev1 commented:
You made sure this is still in a booting state?

SpectreDev1 commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -247,6 +247,18 @@ define Device/redmi_ax6
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	SOC := ipq8072
+	DEVICE_PACKAGES := ipq-wifi-askey_sax1v1k e2fsprogs kmod-fs-ext4 losetup
```
`e2fsprogs kmod-fs-ext4 losetup` you don't need this, it's done by default

gotofbi replied:
fixed and confirmed it compiles and works fine

SpectreDev1 commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -247,6 +247,18 @@ define Device/redmi_ax6
 TARGET_DEVICES += redmi_ax6
 
+define Device/spectrum_sax1v1k
+	$(call Device/FitImage)
```
`$(call Device/EmmcImage)` you need to call this too

gotofbi replied:
fixed and confirmed it compiles and works fine

gotofbi commented:
> You made sure this is still in a booting state?

Initial commits and most of change request were confident enough that its still in a booting state.
However, those two change request you made for ipq807x.mk; I wasn't sure about it so I am recompiling and doing another test.
Will post result once test is done.

gotofbi commented:
> > You made sure this is still in a booting state?
> 
> Initial commits and most of change request were confident enough that its still in a booting state. However, those two change request you made for ipq807x.mk; I wasn't sure about it so I am recompiling and doing another test. Will post result once test is done.

fixed and confirmed it compiles and works fine

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,243 @@
+		port@4 {
+			port_id = <5>;
+			phy_address = <4>;
+		};
```
One port should be removed from whole dts as there is only 3 QCA8075 ports.

gotofbi replied:
fixed!!

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,243 @@
+	switch_wan_bmp = <0x40>; /* wan port bitmap */
+	switch_mac_mode = <0xb>; /* mac mode for uniphy instance0*/
+	switch_mac_mode1 = <0xff>; /* mac mode for uniphy instance1*/
+	switch_mac_mode2 = <0xc>; /* mac mode for uniphy instance2*/
```
You should use names corresponding to this bindings: https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/include/dt-bindings/net/qcom-ipq-ess.h
Example: https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-wpq873.dts#L391-L395

gotofbi replied:
fixed!!

testuser7 commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -247,6 +247,19 @@ define Device/redmi_ax6
 
+define Device/spectrum_sax1v1k
+	$(call Device/FitImage)
+	$(call Device/EmmcImage)
```
`Device/EmmcImage` produce `factory.bin` image that is not used at all.
You can leave this call and add:
    `IMAGES := sysupgrade.bin`
or remove this call and add:
    `IMAGE/sysupgrade.bin/squashfs := append-rootfs | pad-to 64k | sysupgrade-tar rootfs=$$$$@ | append-metadata`

gotofbi replied:
I referenced qnap 301w which is ipq8072 with emmc
```
define Device/qnap_301w
	$(call Device/FitImage)
	$(call Device/EmmcImage)
	DEVICE_VENDOR := QNAP
	DEVICE_MODEL := 301w
	DEVICE_DTS_CONFIG := config@hk01
	KERNEL_SIZE := 16384k
	SOC := ipq8072
	DEVICE_PACKAGES := ipq-wifi-qnap_301w
endef
TARGET_DEVICES += qnap_301w
```

testuser7 replied:
I'm not sure if Qnap uses the factory image. But you don't have to generate an unused image.

gotofbi replied:
fixed

testuser7 commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -247,6 +247,19 @@ define Device/redmi_ax6
+	DEVICE_MODEL := SAX1V1K
+	DEVICE_DTS_CONFIG := config@rt5010w-d187-rev6
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
```
`BLOCKSIZE` and `PAGESIZE` are not used in the case of MMC.

gotofbi replied:
fixed

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,243 @@
+	status = "okay";
+};
+
+&usb_0 {
```
`&qusb_phy_0`, `&ssphy_0` and `&usb_0` should be removed as there is no any USB port.

gotofbi replied:
fixed

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,221 @@
+		qca8075_0: ethernet-phy@0 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <0>;
+		};
```
This port should also be removed.

gotofbi replied:
fixed

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,221 @@
+	switch_lan_bmp = <(ESS_PORT2 | ESS_PORT3 | ESS_PORT4)>; /* lan port bitmap */
+	switch_wan_bmp = <ESS_PORT6>; /* wan port bitmap */
+	switch_mac_mode = <MAC_MODE_QSGMII>; /* mac mode for uniphy instance0*/
+	switch_mac_mode1 = <MAC_MODE_DISABLED>; /* mac mode for uniphy instance1*/
```
Remove this as it's disabled by default.

gotofbi replied:
are you sure to remive "switch_lan_bmp", "switch_wan_bmp", "switch_mac_mode" and "switch_mac_mode1"?

testuser7 replied:
Remove only `switch_mac_mode1`.

gotofbi replied:
fixed!

testuser7 commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -247,6 +247,18 @@ define Device/redmi_ax6
+	DEVICE_MODEL := SAX1V1K
+	DEVICE_DTS_CONFIG := config@rt5010w-d187-rev6
+	SOC := ipq8072
+	IMAGES += sysupgrade.bin
```
Use `:=` not `+=` as you need to define only sysupgrade image.

gotofbi commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,215 @@
+	switch_mac_mode2 = <MAC_MODE_SGMII_PLUS>; /* mac mode for uniphy instance2*/
+
+	qcom,port_phyinfo {
+		port@1 {
```
fixed!!

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,215 @@
+		port@2 {
+			port_id = <2>;
+			phy_address = <1>;
+		};
```
Add spaces after each node.

gotofbi replied:
fixed!!

testuser7 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,218 @@
+
+&sdhc_1 {
+	status = "okay";
+	/delete-property/ mmc-hs400-1_8v;
```
Is `hs400` speed still a problem on the latest sources?

gotofbi replied:
Honestly, I am not sure how to test this.
I referenced mmc portion from QNAP 301W
https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-301w.dts

testuser7 replied:
So add a comment why the property was removed.

gotofbi replied:
fixed!

gotofbi commented:
Once BDF is merged, I will bump ipq-wifi and this PR will be ready to be merged.

testuser7 commented:
> Once BDF is merged, I will bump ipq-wifi and this PR will be ready to be merged.

Can you check the radio on the BDF file with the updated regdb database: https://github.com/testuser7/firmware_qca-wireless/blob/37f16c37b14bb6c8cf66d3854db76a809fe001f5/board-spectrum_sax1v1k.ipq8074 ?

And compare the result of the `iw reg get` command before and after updating the regdb.

gotofbi commented:
> > Once BDF is merged, I will bump ipq-wifi and this PR will be ready to be merged.
> 
> Can you check the radio on the BDF file with the updated regdb database: https://github.com/testuser7/firmware_qca-wireless/blob/37f16c37b14bb6c8cf66d3854db76a809fe001f5/board-spectrum_sax1v1k.ipq8074 ?
> 
> And compare the result of the `iw reg get` command before and after updating the regdb.

Im not fully understanding how to update regdb

```
root@OpenWrt:/# opkg install wireless-regdb
Package wireless-regdb (2024.01.23-1) installed in root is up to date.
```
Looks I guess current snapshot comes with latest wireless-regdb?

```
root@OpenWrt:/# iw reg get
global
country 00: DFS-UNSET
        (755 - 928 @ 2), (N/A, 20), (N/A), PASSIVE-SCAN
        (2402 - 2472 @ 40), (N/A, 20), (N/A)
        (2457 - 2482 @ 20), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
        (2474 - 2494 @ 20), (N/A, 20), (N/A), NO-OFDM, PASSIVE-SCAN
        (5170 - 5250 @ 80), (N/A, 20), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (N/A, 20), (0 ms), DFS, AUTO-BW, PASSIVE-SCAN
        (5490 - 5730 @ 160), (N/A, 20), (0 ms), DFS, PASSIVE-SCAN
        (5735 - 5835 @ 80), (N/A, 20), (N/A), PASSIVE-SCAN
        (57240 - 63720 @ 2160), (N/A, 0), (N/A)

phy#1 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5835 @ 80), (6, 30), (N/A), AUTO-BW

phy#0 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5835 @ 80), (6, 30), (N/A), AUTO-BW
```

This is current result

testuser7 commented:
> Im not fully understanding how to update regdb
> 

Use the board file from the provided repository and then place it in the path: `/lib/firmware/ath11k/IPQ8074/hw2.0/board-2.bin`
You can restart your router after that.

gotofbi commented:
> > Im not fully understanding how to update regdb
> 
> Use the board file from the provided repository and then place it in the path: `/lib/firmware/ath11k/IPQ8074/hw2.0/board-2.bin` You can restart your router after that.

Oh my bad, 
I thought the link you provided was same file that I created PR.
Compiling new build atm, will post result shortly.

gotofbi commented:
> root@OpenWrt:/# iw reg get
> global
> country 00: DFS-UNSET
>         (755 - 928 @ 2), (N/A, 20), (N/A), PASSIVE-SCAN
>         (2402 - 2472 @ 40), (N/A, 20), (N/A)
>         (2457 - 2482 @ 20), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
>         (2474 - 2494 @ 20), (N/A, 20), (N/A), NO-OFDM, PASSIVE-SCAN
>         (5170 - 5250 @ 80), (N/A, 20), (N/A), AUTO-BW
>         (5250 - 5330 @ 80), (N/A, 20), (0 ms), DFS, AUTO-BW, PASSIVE-SCAN
>         (5490 - 5730 @ 160), (N/A, 20), (0 ms), DFS, PASSIVE-SCAN
>         (5735 - 5835 @ 80), (N/A, 20), (N/A), PASSIVE-SCAN
>         (57240 - 63720 @ 2160), (N/A, 0), (N/A)
> 
> phy#1 (self-managed)
> country US: DFS-FCC
>         (2402 - 2472 @ 40), (6, 30), (N/A)
>         (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
>         (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
>         (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
>         (5735 - 5835 @ 80), (6, 30), (N/A), AUTO-BW
> 
> phy#0 (self-managed)
> country US: DFS-FCC
>         (2402 - 2472 @ 40), (6, 30), (N/A)
>         (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
>         (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
>         (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
>         (5735 - 5835 @ 80), (6, 30), (N/A), AUTO-BW

Result after upgrade.

```
root@OpenWrt:/# iw reg get
global
country 00: DFS-UNSET
        (755 - 928 @ 2), (N/A, 20), (N/A), PASSIVE-SCAN
        (2402 - 2472 @ 40), (N/A, 20), (N/A)
        (2457 - 2482 @ 20), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
        (2474 - 2494 @ 20), (N/A, 20), (N/A), NO-OFDM, PASSIVE-SCAN
        (5170 - 5250 @ 80), (N/A, 20), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (N/A, 20), (0 ms), DFS, AUTO-BW, PASSIVE-SCAN
        (5490 - 5730 @ 160), (N/A, 20), (0 ms), DFS, PASSIVE-SCAN
        (5735 - 5835 @ 80), (N/A, 20), (N/A), PASSIVE-SCAN
        (57240 - 63720 @ 2160), (N/A, 0), (N/A)

phy#1 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW

phy#0 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW
```

I notice this is updated for phy#1 and phy#0
```
(5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW
```

testuser7 commented:
> I notice this is updated for phy#1 and phy#0
> 
> ```
> (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW
> ```
> 
So you can try running HE160 on channel 149 with this PR: https://github.com/openwrt/openwrt/pull/14926
Set the country to `US`.

gotofbi commented:
> > I notice this is updated for phy#1 and phy#0
> > ```
> > (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW
> > ```
> 
> So you can try running HE160 on channel 149 with this PR: #14926 Set the country to `US`.

Runs fine and I am able to connect wifi.
Tested with iPhone 13Pro and M1 Macbook Pro

However, when I checked "Associated Stations" both iPhone and M1 Macbook Pro are connected as "80 MHz"
From Macbook, wifi diagnostic shows that I am also connected Channel: 149 (5 GHz, 80 MHz)

testuser7 commented:
> However, when I checked "Associated Stations" both iPhone and M1 Macbook Pro are connected as "80 MHz" From Macbook, wifi diagnostic shows that I am also connected Channel: 149 (5 GHz, 80 MHz)

Can you check supported channels: https://apple.stackexchange.com/a/74575 ?

gotofbi commented:
> > However, when I checked "Associated Stations" both iPhone and M1 Macbook Pro are connected as "80 MHz" From Macbook, wifi diagnostic shows that I am also connected Channel: 149 (5 GHz, 80 MHz)
> 
> Can you check supported channels: https://apple.stackexchange.com/a/74575 ?

System report says it support 149 5GHz but doesnt say if it supports 160MHz or not.
However, since im in Canada, current Country Code is : CA for the laptop.

testuser7 commented:
@gotofbi Your commit should include all information provided in the PR description and should additionally be marked as `Signed-off-by`.

testuser7 commented:
> System report says it support 149 5GHz but doesnt say if it supports 160MHz or not. However, since im in Canada, current Country Code is : CA for the laptop.
> 
> Edit 1: https://support.apple.com/en-ie/guide/deployment/dep2ac3e3b51/web Doesnt look like my laptop supports 160MHz
> 
> Edit 2: Forgot that I have desktop with Intel AX200 CH 36 : 80MHz works CH 36 : 160MHz doesnt work (wifi is not visble on desktop nor other mobile devices)
> 
> CH149 : 160MHz partially works. it appears on both desktop and mobile device, when I connect from desktop, I am connected as 802.11n mode only 40MHz with max 300Mbps CH149 : 80MHz works

Does 80MHz work on channel 165?

gotofbi commented:
> > System report says it support 149 5GHz but doesnt say if it supports 160MHz or not. However, since im in Canada, current Country Code is : CA for the laptop.
> > Edit 1: https://support.apple.com/en-ie/guide/deployment/dep2ac3e3b51/web Doesnt look like my laptop supports 160MHz
> > Edit 2: Forgot that I have desktop with Intel AX200 CH 36 : 80MHz works CH 36 : 160MHz doesnt work (wifi is not visble on desktop nor other mobile devices)
> > CH149 : 160MHz partially works. it appears on both desktop and mobile device, when I connect from desktop, I am connected as 802.11n mode only 40MHz with max 300Mbps CH149 : 80MHz works
> 
> Does 80MHz work on channel 165?

Works but looks like its partially broken.
Wifi is functioning.
However, both iPhone and AX200 are connected as 20MHz.
AX200 reports that its connected as 802.11n

testuser7 commented:
> Works but looks like its partially broken. Wifi is functioning. However, both iPhone and AX200 are connected as 20MHz. AX200 reports that its connected as 802.11n

You can also check the channels individually with the `HE20` option: 169, 173, 177

gotofbi commented:
> > Works but looks like its partially broken. Wifi is functioning. However, both iPhone and AX200 are connected as 20MHz. AX200 reports that its connected as 802.11n
> 
> You can also check the channels individually with the `HE20` option: 169, 173, 177

`HE20` 165 Works
`HE20` 169, 173, 177 Not visible from other device.

testuser7 commented:
> `HE20` 165 Works `HE20` 169, 173, 177 Not visible from other device.

Have you set your country to `US`? 165 is the last channel for an unset country code (`00`).

testuser7 commented:
> Correct, I confirmed that country code is set to 'US' in OpenWrt I used board file you posted on your git.

Can you post the output of `iw reg get` now?

gotofbi commented:
> > Correct, I confirmed that country code is set to 'US' in OpenWrt I used board file you posted on your git.
> 
> Can you post the output of `iw reg get` now?

```
root@OpenWrt:/# iw reg get
global
country US: DFS-FCC
        (902 - 904 @ 2), (N/A, 30), (N/A)
        (904 - 920 @ 16), (N/A, 30), (N/A)
        (920 - 928 @ 8), (N/A, 30), (N/A)
        (2400 - 2472 @ 40), (N/A, 30), (N/A)
        (5150 - 5250 @ 80), (N/A, 23), (N/A), AUTO-BW
        (5250 - 5350 @ 80), (N/A, 24), (0 ms), DFS, AUTO-BW
        (5470 - 5730 @ 160), (N/A, 24), (0 ms), DFS
        (5730 - 5850 @ 80), (N/A, 30), (N/A), AUTO-BW
        (5850 - 5895 @ 40), (N/A, 27), (N/A), NO-OUTDOOR, AUTO-BW, PASSIVE-SCAN
        (5925 - 7125 @ 320), (N/A, 12), (N/A), NO-OUTDOOR, PASSIVE-SCAN
        (57240 - 71000 @ 2160), (N/A, 40), (N/A)

phy#1 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW

phy#0 (self-managed)
country US: DFS-FCC
        (2402 - 2472 @ 40), (6, 30), (N/A)
        (5170 - 5250 @ 80), (6, 30), (N/A), AUTO-BW
        (5250 - 5330 @ 80), (6, 24), (0 ms), DFS, AUTO-BW
        (5490 - 5730 @ 160), (6, 24), (0 ms), DFS, AUTO-BW
        (5735 - 5895 @ 160), (6, 30), (N/A), AUTO-BW

```

testuser7 commented:
So probably the hardware or firmware does not allow the use of these frequencies.

motolav commented:
My Dynalink has the exact same `iw reg get` output, using snapshot with the latest ath11k firmware

gotofbi commented:
> @gotofbi Your commit should include all information provided in the PR description and should additionally be marked as `Signed-off-by`.

fixed!

gotofbi commented:
> Hi sorry for hijacking this thread, but can you share /proc/cpuinfo of this device? I'm interested on CPU Implementer and CPU part, thanks!

Its pretty much identical as Dynalink-DL-WRX36
Minor details are different (missing usb, 1gb emmc instead of 256mb nand, 2gb ram instead of 1gb, 1less eth port)
Both from same manufacture, same model, different sub model (Askey RT5010W)

```
root@OpenWrt:~# cat /proc/cpuinfo
processor       : 0
BogoMIPS        : 38.40
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 1
BogoMIPS        : 38.40
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 2
BogoMIPS        : 38.40
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 3
BogoMIPS        : 38.40
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4
```

gotofbi commented:
@robimarko @testuser7 
bump for visibility.

testuser7 commented:
@gotofbi Someone on the forum mentions problems with sysupgrade: https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923

gotofbi commented:
> @gotofbi Someone on the forum mentions problems with sysupgrade: https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923

I am almost sure that the user didn't follow instruction.
After sysupgrade, it requires Factory reset to clean up the partition.

https://forum.openwrt.org/t/spectrum-sax1v1k-askey-rt5010w-openwrt-support/149923/114
Its mentioned on this thread.
Also, I mentioned on Installation Instruction that it needs factory reset on step 6.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-sax1v1k.dts:
```diff
@@ -0,0 +1,223 @@
+&switch {
+	status = "okay";
+
+	switch_cpu_bmp = <ESS_PORT0>;  /* cpu port bitmap */
```
This is already the default in ESS DTSI

gotofbi replied:
fixed!!

gotofbi commented:
> I have prepared a PR with an updated regdb in a BDF: [openwrt/firmware_qca-wireless#39](https://github.com/openwrt/firmware_qca-wireless/pull/39)

I see original PR for BDF has been merged.
However, I will wait your updated BDF to be merged and bump ipq-wifi

robimarko commented:
Please sort out the formal issue and update ipq-wifi to include updated BDF as well as rebase this and it should be ready to go

gotofbi commented:
> Please sort out the formal issue and update ipq-wifi to include updated BDF as well as rebase this and it should be ready to go

 If you referring formal issue as the one that posted on the forum; its user error which failed to follow instruction of hard-reset after flash.
 I checked about updating ipq-wifi package but @aiamadeus already updated it on this commit
 https://github.com/openwrt/openwrt/commit/9cef2bc224b9bda19a8a79df6d1de9560c081ce2
 Lastly, I rebased to head and believe its ready to go finally :)

robimarko commented:
I mean, CI formal issue:
https://github.com/openwrt/openwrt/actions/runs/8783741000/job/24111662004

Yes, the ipq-wifi package was updated for another device so its fine.

gotofbi commented:
> I mean, CI formal issue: https://github.com/openwrt/openwrt/actions/runs/8783741000/job/24111662004
> 
> Yes, the ipq-wifi package was updated for another device so its fine.

I think it failed due to incorrect github profile that I had.
I updated my profile and hope it unblocks CI.

robimarko commented:
Github profile doesn't have anything to do with this, error is quite clear that commit author does not match the Signed-off-by, please update it by: `git commit --ammend --author="Connor Yoon <j_connor@taliaent.com>"` and force push

gotofbi commented:
> Github profile doesn't have anything to do with this, error is quite clear that commit author does not match the Signed-off-by, please update it by: `git commit --ammend --author="Connor Yoon <j_connor@taliaent.com>"` and force push

my bad and thanks for guiding for solution.
updated author and force pushed.

robimarko commented:
Thanks! Rebased on top of main and merged!

dangerweenie commented:
Hi there!

After running the directions above (in the first post - regarding connecting to the router via serial), I got the following error (when trying to run open.sh).

To be clear I haven't done anything except connected to the router and logged in as root - I'm not sure if I am supposed to be running these commands after some other steps, modfications, or flashing?

`root@SAX1V1K:~# vi
root@SAX1V1K:~# vi open.sh
root@SAX1V1K:~# vi open.sh
root@SAX1V1K:~# chmod 755 open.sh
root@SAX1V1K:~# ./open.sh
Finding partitions..
Dumping patch locations..
4308de5
b3c974a
Cant proceed, unknown U-Boot binary! Please send a hexdump of /dev/mmcblk0p15 so I can check it.
root@SAX1V1K:~#
`

dangerweenie commented:
> My Dynalink has the exact same `iw reg get` output, using snapshot with the latest ath11k firmware

are you able to use the latest snapshot?  I can only use 2242 as indicated in the guide - anything newer, WAN doesnt connect.
